lib/tether/ops_linux.go

```diff
@@ -699,6 +700,52 @@ func (t *BaseOperations) MountTarget(ctx context.Context, source url.URL, target
 	return nil
 }
 
+// CopyExistingContent copies the underlying files shadowed by a mount on a directory
+// to the volume mounted on the directory
+// see bug https://github.com/vmware/vic/issues/3482
+func (t *BaseOperations) CopyExistingContent(source string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("copyExistingContent from %s", source)))
+
+	source = filepath.Clean(source)
+
+	log.Debugf("creating directory %s", bindDir)
+	if err := os.MkdirAll(bindDir, 0644); err != nil {
+		log.Errorf("error creating directory %s: %+v", bindDir, err)
+		return err
+	}
+
+	parentDir := filepath.Dir(source)
+	// mount the parent directory of the source to bindDir
+	// e.g if source is /foo/bar, mount /foo to ./bindDir
+	log.Debugf("mounting %s on %s", parentDir, bindDir)
+	if err := Sys.Syscall.Mount(parentDir, bindDir, ext4FileSystemType, syscall.MS_BIND, ""); err != nil {
+		log.Errorf("error mounting to %s: %+v", bindDir, err)
+		return err
+	}
+
+	mountedSource := filepath.Join(bindDir, filepath.Base(source))
+	// copy data from the bindDir to the source
+	// e.g if source is /foo/bar, copy ./bindDir/bar to /foo/bar
+	log.Debugf("copying contents from to %s to %s", mountedSource, source)
+	if err := archive.CopyWithTar(mountedSource, source); err != nil {
```

Comment:
Isn't this an explicit copy?  It will copy over the dest even if the dest exists, no?  We really don't want that.  Infact, we shouldn't be copying at all if anything exists in the volume (except for the `lost+found` directory).